cl/citations/tests.py

```diff
@@ -998,6 +1084,82 @@ def test_no_duplicate_parentheticals_from_parallel_cites(self) -> None:
+        self.assertIsInstance(result[0], OpinionDocument)
+
+        # Validate that we matched the pincite to the correct cluster
+        self.assertEqual(result[0]["cluster_id"], self.citation6.cluster.id)
+
+        # Can we match the pincite when we have multiple opinions?
+        pincite2 = case_citation(
+            volume="9",
+            reporter="Pet.",
+            page="745",
+        )
+        result, citation_found = es_search_db_for_full_citation(pincite2)
+
+        # Find the second opinion
+        concurrence_opinion = Opinion.objects.get(
+            cluster=self.cluster1, type="030concurrence"
+        )
+
+        # Verify that we found a match
+        self.assertEqual(citation_found, True)
+        # Verify that the result is from the correct opinion
+        self.assertEqual(
+            result[0]["id"],
+            concurrence_opinion.pk,
+            "Matched in the wrong opinion.",
+        )
+        # Verify the correct matched cluster
+        self.assertEqual(
+            result[0]["cluster_id"], self.cluster1.pk, "Matched wront cluster"
+        )
+
+        # Verify that we can annotate pin cites correctly
+        opinion = Opinion.objects.get(cluster=self.citation8.cluster)
+        get_and_clean_opinion_text(opinion)
+        # Extract the citations from the opinion's text
+        citations = get_citations(
+            opinion.cleaned_text, tokenizer=HYPERSCAN_TOKENIZER
+        )
+        citation_resolutions = do_resolve_citations(citations, opinion)
+
+        matched_opinion = Opinion.objects.get(cluster=self.citation7.cluster)
+
+        # Generate the citing opinion's new HTML with inline citation links
+        new_html = create_cited_html(opinion, citation_resolutions)
+
+        main_citation = f"""<span class="citation" data-id="{matched_opinion.pk}"><a href="/opinion/{self.citation7.cluster.pk}/united-states-v-paramount-pictures-inc/" aria-description="Citation for case: United States v. Paramount Pictures, Inc.">334 U. S. 131</a></span>"""
+        pincite_1 = f"""<span class="citation pin-cite" data-id="{matched_opinion.pk}"><a href="/opinion/{self.citation7.cluster.pk}/united-states-v-paramount-pictures-inc/#144" aria-description="Citation for case: United States v. Paramount Pictures, Inc.">144-145, n. 6</a></span>"""
+        pincite_2 = f""" <span class="citation pin-cite" data-id="{matched_opinion.pk}"><a href="/opinion/{self.citation7.cluster.pk}/united-states-v-paramount-pictures-inc/#144" aria-description="Citation for case: United States v. Paramount Pictures, Inc.">144, n. 6</a></span> (1948).</p>"""
+
+        # Verify that a citation without pin cite matches
+        self.assertIn(main_citation, new_html)
+        # Verify that a citation with pin cite with single page number matches
+        self.assertIn(pincite_1, new_html)
+        # Verify that a citation with pin cite with page range matches
+        self.assertIn(pincite_2, new_html)
+
```

Comment:
can this be refactored and incorporated into 

class CitationTextTest(SimpleTestCase):
    def test_make_html_from_plain_text(self) -> None:
    

I think that would make this much easier for me to understand.

Reply:
I updated the tests